src/js/jsx/sections/layers/LayersGroup.jsx

```diff
@@ -0,0 +1,103 @@
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React),
+        classnames = require("classnames");
+
+    var DummyLayerFace = require("./DummyLayerFace"),
+        LayerFace = require("./LayerFace");
+
+    var LayersGroup = React.createClass({
+        mixins: [FluxMixin],
+
+        getDefaultProps: function () {
+            return {
+                depth: 0,
+                isRoot: false
+            };
+        },
+
+        render: function () {
+            var layerFaces = this.props.layerNodes.reduce(function (results, layerNode) {
+                var layer = this.props.document.layers.byID(layerNode.id);
+                
+                if (!layer.isGroupEnd) {
+                    var layersGroup = layerNode.children && (
+                        <LayersGroup
+                            disabled={this.props.disabled}
+                            document={this.props.document}
+                            layerNodes={layerNode.children}
+                            depth={this.props.depth + 1}/>
+                    );
+
+                    var className = classnames({
+                        "layer-group": layersGroup,
+                        "layer-group__selected": layersGroup && layer.selected,
+                        "layer-group__collapsed": layersGroup && !layer.expanded,
+                        "layer-group__not-visible": layersGroup && !layer.visible
+                    });
+
+                    results.push(
+                        <li key={layer.key} className={className}>
+                            <LayerFace
+                                key={layer.key}
+                                disabled={this.props.disabled}
+                                document={this.props.document}
+                                depth={this.props.depth}
+                                layer={layer}
+                                childNodes={layerNode.children}/>
+                            {layersGroup}
+                        </li>
+                    );
+                }
+                
+                return results;
+            }.bind(this), []);
+
+            if (this.props.isRoot) {
```

Comment:
Yes, only the root layer group has the dummy layer face. I am not sure exposing the `DummyLayerFace` to the `LayerPanel` is a good idea, as they seem unrelated, besides, this will require the `LayerGroup` to provide a way to allow appending layer face to the list, which will add extra complexity.